### PR TITLE
feat: Add semantic-release and configure

### PR DESCRIPTION
**Pull request by atolivero**
Do this to enable npm and GH package publishing

### CODE REVIEW

**Comment by atolivero** [comment]
Since we need to do the GH publish in addition to NPM, we can't rely on `npx lifeomic-publish-npm-package --publish-tagged-commits-only` from an `on: push: branches: - 'master'` action.  And `on: push: tag` actions are wonky to configure to be restricted to master branch only.  

I think utilizing `semantic-release` here is the easiest path forward for this publish.

**Review by alexccl — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -17,7 +17,8 @@
     "lint": "eslint .",
     "pretest": "yarn lint",
     "test": "ava --serial --verbose test/unit",
-    "build": "./build.sh"
+    "build": "./build.sh",
+    "semantic-release": "semantic-release"
```
Comment (smolster) [inline package.json:21]: I think this line could be removed entirely

File: release.config.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+module.exports = {
+  branches: ['master'],
+  plugins: [
```
Comment (smolster) [inline release.config.js:3]: I'd recommend making these changes:
1. `yarn add -D conventional-changelog-conventionalcommits`
2. add this entry to the beginning of this plugin list:
    `['@semantic-release/commit-analyzer', { preset: 'conventionalcommits' }],`
3. Create a new file at `.github/lifeomic-probot.yml` that looks like this:
    ```yml
    enforceSemanticCommits: true
    ```

Why the probot thing: the probot config will perform nice checks on PRs to ensure every PR contains a semantic commit, and flag major version bumps.

Why "conventional-commits":
- every other project in our org that uses `semantic-release` uses this commit spec, AFAIK "conventional commits" spec, and
- the probot config currently only supports that spec

**Review by smolster — COMMENTED** [review]
I'd also recommend setting the version in `package.json` to `0.0.0` -- this is a convention we've adopted throughout the org. Helps to communicate that the version in-code is meaningless.

**Comment by smolster** [comment]
It's fine to merge this as-is and make the changes I suggested in a subsequent PR.

**Pull request merged**